### ISSUE TITLE
JupyterLab 1.1.x backwards incompatibility

### ISSUE THREAD

**Issue by jasongrout** [jupyterlab/jupyterlab#7181]
We've been seeing a lot of issues about 1.0.x with extensions installed breaking after 1.1 came out. Here's a reproducible setup:

```sh
conda create -n jlab107 jupyterlab=1.0.7
conda activate jlab107
jupyter labextension install @jupyterlab/toc
jupyter lab build
jupyter lab
```
Note that we built a second time after installing the extension.

Now create a new notebook, and notice the tabs are off and there are no icons for the sidebar panels:

<img width="1164" alt="Screen Shot 2019-09-09 at 3 47 04 PM" src="https://user-images.githubusercontent.com/192614/64571570-2599f280-d319-11e9-85d0-504d9b1eb1fe.png">

This also happens if, for example, we replace installing `@jupyterlab/toc` with installing `@jupyter-widgets/jupyterlab-manager`.

A relevant error in the js console seems to be from this line:

https://github.com/jupyterlab/jupyterlab/blob/b80c738f5a89bfcbfea6a23ac7d808a069e0bba0/packages/apputils/src/thememanager.ts#L283

The error is: "TypeError: this._settings.schema.definitions is undefined"

A guess at what is happening: In 1.0.7, the dependencies were `^1.0`, so it is pulling in packages from jlab 1.1. In 1.0.8, we accidentally introduced yarn resolutions which pinned the packages to 1.0.x packages, so we don't get 1.1 packages and the error doesn't occur. Those yarn resolutions propagated all the way through 1.0.10, the latest patch release of 1.0.

However, it seems like you will still have this issue in 1.0.x for `x<=7`.

CC @telamonian

**Comment by jasongrout** [comment]
Setting as 1.2, but a fix should also be backported to 1.1.

**Comment by telamonian** [comment]
I found at least 3 related problems that could mess up Jlab's appearance if certain packages are out of sync and newer than the others. 

The worst of these (the one responsible for the error boxes popping up in place of icons) is due to the fact that the latest version of `ui-components` relies on some corresponding changes made to how Webpack loads SVGs in `webpack.config.js`. The best fix for this would be to backport the SVG related changes made to `webpack.config.js` to the `1.0.x` branch. Can we do that? How should I go about it?

It might also be possible to fix by detecting the error and forcing a fallback to the old icon handling behavior in all cases. This would be at least a little involved, and it doesn't make a ton of sense to me to make the latest version more convoluted in order to fix a bug in an outdated version.

**Comment by blink1073** [comment]
The underlying issue is that 1.0.2 is the version in the default anaconda channel.  if we cut a new 1.0.x it won't help folks in that boat.

**Comment by blink1073** [comment]
I am taking a step back from OSS for a week, I need a mental recharge.

**Comment by telamonian** [comment]
Hmm, in that case I'll come up with something. I'm prepping a PR right now

**Comment by jasongrout** [comment]
@blink1073 - thanks for your huge push this last little while!

@telamonian - 1.08, 1.09, and 1.0.10 inadvertently have a fix (they won't pull in 1.1.x packages). We need to fix the issue for 1.0.x for `x<=7`, which means we need to release 1.1.x packages that work with them. The workarounds only need to be active in the 1.x cycle, though, so 2.0 in January can have the clean new code.

**Comment by telamonian** [comment]
The PR exists (#7182)!

Overall, this turned out to be a pretty good opportunity to improve error handling in `IconRegistry` in the case of bad SVG html. For the other problems, I found some reasonable hacks, and added some `TODO`s to remind to delete said hacks in `v2.0.0`
